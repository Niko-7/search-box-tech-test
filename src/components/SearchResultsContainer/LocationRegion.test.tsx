/*eslint-disable */
import { cleanup, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import { LocationRegion } from ".";
const { axe } = require("jest-axe");
import "jest-axe/extend-expect";

afterEach(() => {
  cleanup();
});

describe("LocationRegion component", () => {
  it("should render the region and country in a single string seperated by a comma", () => {
    render(
      <LocationRegion
        region={"Greater Manchester"}
        country={"United Kingdom"}
        city={"Manchester"}
      />
    );
    const locationRegion = screen.getByText(
      "Manchester, Greater Manchester, United Kingdom"
    );
    expect(locationRegion).toBeInTheDocument;
  });
});

describe("Snapshot tests", () => {
  it("should match the snapshot of the LocationRegion component", () => {
    const tree = renderer
      .create(
        <LocationRegion
          region={"Greater Manchester"}
          country={"United Kingdom"}
          city={"Manchester"}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("accesibility tests", () => {
  it("should have no accesibility errors on LocationRegion component", async () => {
    const { container } = render(
      <LocationRegion
        region={"Greater Manchester"}
        country={"United Kingdom"}
        city={"Manchester"}
      />
    );
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
