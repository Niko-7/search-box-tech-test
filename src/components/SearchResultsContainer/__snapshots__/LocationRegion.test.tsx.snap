// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests should match the snapshot of the LocationRegion component 1`] = `
<div
  className="search-results__location-region"
>
  Manchester, Greater Manchester, United Kingdom
</div>
`;
